plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/DecoratedHiveMetastoreModule.java

```diff
@@ -42,6 +42,18 @@
 public class DecoratedHiveMetastoreModule
         extends AbstractConfigurationAwareModule
 {
+    private final boolean installFlushMetadataCacheProcedure;
```

Comment:
the flush procedure should be installed whenever caching is supported

what about changing the parameter name and logic.
for example, we could skip binding `CachingHiveMetastoreConfig` for iceberg (and maybe some others)

Reply:
> the flush procedure should be installed whenever caching is supported

This makes it hard to exclude Hive's `flush_metadata_cache` from Delta Lake connector. I initially tried to exclude the procedure from `DeltaLakeConnector#getProcedures`, but `Procedure` class doesn't contain catalog information.